src/Raven.Server/RavenServer.cs

```diff
@@ -284,7 +301,119 @@ void ConfigureKestrel(KestrelServerOptions options)
                     Logger.Operations("Could not start server", e);
                 throw;
             }
+        }
+
+        public MultipleUseFlag CpuCreditsAlertRaised = new MultipleUseFlag();
+        public CpuCreditsState CpuCreditsBalance = new CpuCreditsState();
+        public class CpuCreditsState : IDynamicJson
+        {
+            public bool Used;
+            public double BaseCredits;
+            public double MaxCredits;
+            public double Threshold;
+            public double RemainingCpuCredits;
+            public double ThresholdReleasValue;
+            public double CreditsAccuredPerSecond;
+            public double[] History = new double[60 * 60];
+            public int HistoryCurrentIndex;
+
+            public DynamicJsonValue ToJson()
+            {
+                var historyByMinute = new DynamicJsonArray();
+                var current = HistoryCurrentIndex;
+                int currentMinItems = 0;
+                var currentMinuteValue = 0d;
+                for (int i = 0; i < History.Length; i++)
+                {
+                    var val = History[(current + i) % History.Length];
+                    currentMinuteValue += val;
+                    if(++currentMinItems == 60)
+                    {
+                        currentMinItems = 0;
+                        historyByMinute.Add(currentMinuteValue / 60);
+                        currentMinuteValue = 0d;
+                    }
+                }
+
+                return new DynamicJsonValue
+                {
+                    [nameof(Used)] = Used,
+                    [nameof(BaseCredits)] = BaseCredits,
+                    [nameof(MaxCredits)] = MaxCredits,
+                    [nameof(Threshold)] = Threshold,
+                    [nameof(RemainingCpuCredits)] = RemainingCpuCredits,
+                    [nameof(ThresholdReleasValue)] = ThresholdReleasValue,
+                    [nameof(CreditsAccuredPerSecond)] = CreditsAccuredPerSecond,
```

Comment:
typo: `CreditsAccuredPerSecond`... `CreditsOccuredPerSecond`? `CreditsAquiredPerSecond`? Not sure what you meant.